src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,903 @@
+
+template<int dim, int nstate>
+double AdvectionPeriodic<dim, nstate>::compute_energy(std::shared_ptr < PHiLiP::DGBase<dim, double> > &dg) const
+{
+	double energy = 0.0;
+        dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->right_hand_side);
+        dg->global_mass_matrix.vmult( Mu_hat, dg->solution);
+        energy = dg->solution * Mu_hat;
+	return sqrt(energy);
+}
+
+template<int dim, int nstate>
+double AdvectionPeriodic<dim, nstate>::compute_conservation(std::shared_ptr < PHiLiP::DGBase<dim, double> > &dg, const double poly_degree) const
+{
+	double conservation = 0.0;
+       // dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->solution.size());
+        dealii::LinearAlgebra::distributed::Vector<double> Mu_hat(dg->right_hand_side);
+       // printf(" size Mass %d size sol %d\n",dg->global_mass_matrix.n(),dg->solution.size());
+       // printf("  Mass %g size sol %g\n",dg->global_mass_matrix(dg->global_mass_matrix.m(),dg->global_mass_matrix.n()),dg->solution(dg->solution.size()));
+        //dg->global_mass_matrix.vmult( Mu_hat, dg->right_hand_side);
+        dg->global_mass_matrix.vmult( Mu_hat, dg->solution);
+
+        const unsigned int n_dofs_cell = dg->fe_collection[poly_degree].dofs_per_cell;
+        const unsigned int n_quad_pts = dg->volume_quadrature_collection[poly_degree].size();
+        dealii::FullMatrix<double> Chi_operator(n_quad_pts, n_dofs_cell);
+        for(int istate=0; istate<nstate; istate++){
+            for (unsigned int itest=0; itest<n_dofs_cell; ++itest) {
+                for (unsigned int iquad=0; iquad<n_quad_pts; ++iquad) {
+                    const dealii::Point<dim> qpoint  = dg->volume_quadrature_collection[poly_degree].point(iquad);
+                    Chi_operator[iquad][itest] = dg->fe_collection[poly_degree].shape_value_component(itest,qpoint,istate);
+                }
+            }
+        }
+        dealii::FullMatrix<double> Chi_inv(n_dofs_cell);
+        Chi_inv.invert(Chi_operator);
+        dealii::Vector<double> ones(n_quad_pts);
+        for(unsigned int iquad=0; iquad<n_quad_pts; iquad++){
+            ones[iquad] = 1.0;
+        }
+        dealii::Vector<double> ones_hat(n_dofs_cell);
+        Chi_inv.vmult(ones_hat, ones);
+
+        dealii::LinearAlgebra::distributed::Vector<double> ones_hat_global(dg->right_hand_side);
+        std::vector<dealii::types::global_dof_index> dofs_indices (n_dofs_cell);
+        for (auto cell = dg->dof_handler.begin_active(); cell!=dg->dof_handler.end(); ++cell) {
+            if (!cell->is_locally_owned()) continue;
+            cell->get_dof_indices (dofs_indices);
+            for(unsigned int idof=0;idof<n_dofs_cell; idof++){
+                ones_hat_global[dofs_indices[idof]] = ones_hat[idof];
+            }
+        }
+
+        conservation = ones_hat_global * Mu_hat;
+
+	return conservation;
+}
 
 template <int dim, int nstate>
 int AdvectionPeriodic<dim, nstate>::run_test() const
```

Comment:
The indentation is all over the place in thise file, please fix.

The presence of old code in comments might be what's causing the indentation to "reset" here and there.

Reply:
fixed